### DISCUSSION TITLE
Help：about the generation of concepts for MIMIC-IV-2.2

### DISCUSSION THREAD

**Discussion by washingtonwuyi** [MIT-LCP/mimic-code#1627] [MIMIC-IV] [answered]
Hello everyone! It's me agian.Today I tried to generate concepts for my local MIMIC-IV-2.2 build.Following the README instructions,I executed the SQL files on the SQL Shell,and every steps seemed to go normal.However,after the generation of the concepts,I found that the views or the materialized views were under the category of "tables" but not the "views" or "materialized views",no matter the GUI I used was "pgAdmin 4" or "Navicat".Why's that?Any solutions or suggestions?Thank you!
![微信图片编辑_20230821200245](https://github.com/MIT-LCP/mimic-code/assets/136318775/f7c1a675-13c5-4cc9-a157-e7884207721d)
![微信图片编辑_20230821200222](https://github.com/MIT-LCP/mimic-code/assets/136318775/b4b7ebaa-02fd-4678-b526-90a93005aac7)
![微信图片编辑_20230821200144](https://github.com/MIT-LCP/mimic-code/assets/136318775/73e470a2-d555-4255-8300-0d914345332c)
![微信图片编辑_20230821200108](https://github.com/MIT-LCP/mimic-code/assets/136318775/28e345e3-35c6-4f87-9e93-facd8f582674)
![b52fcc044281c8d581425478202da84](https://github.com/MIT-LCP/mimic-code/assets/136318775/051f92c1-2deb-4904-9cea-1ebe9c70c4b4)
![2048909f1c4ede751e3aca19004fbd7](https://github.com/MIT-LCP/mimic-code/assets/136318775/902e633e-8b8a-40f9-9368-72c1a439b1bb)

**Comment by alistairewj** [comment, marked as answer]
This is because... they are tables!

More specifically, these *used* to be materialized views, but somewhere along the way I realized it would be better to simply create tables on a separate schema, rather than materialized views in the same schema. It also makes the code more portable as some database systems don't support materialized views.

You can see here where we echo "creating materialized views":
https://github.com/MIT-LCP/mimic-code/blob/78f7b6e5a8886f52066b0505c451597590bea862/mimic-iv/concepts_postgres/postgres-make-concepts.sql#L3

But then all the scripts actually make tables:
https://github.com/MIT-LCP/mimic-code/blob/78f7b6e5a8886f52066b0505c451597590bea862/mimic-iv/concepts_postgres/demographics/icustay_detail.sql#L1-L2

So this is just a typo/bug in the SQL script.

**Reply by washingtonwuyi** [reply]
Wow，thank you for your answer! My confusion has gone after reading your explanation.Thank you for your time and patience.Have a nice day!
